sw/airborne/mcu_periph/uart.h

```diff
@@ -36,8 +36,13 @@
 #define UART_RX_BUFFER_SIZE 128
 #endif
 #ifndef UART_TX_BUFFER_SIZE
+#ifdef STM32F4  //the F4 has enough memory, and the PX4 bootloader needs more then 128
```

Comment:
I'm really not in favour of increasing default buffer size, it's not needed for 99% of use cases and on some of the low end LPCs/F1s not even possible to to ram size.

Defining a bigger buffer for e.g. linux or certain STMs could be done in their respective `uart_arch.h` files.